### PR TITLE
Added support for NOT RLIKE expressions

### PR DESCRIPTION
**Pull request by guandatawangjialin**
Added support for NOT RLIKE expressions
#1553

### CODE REVIEW

**Comment by guandatawangjialin** [comment]
#1553

**Comment by manticore-projects** [comment]
Greetings!

Thank you very much for your contribution.
Although I have some concerns:

1) You extend the `LikeExpression`, but without defining what variant of `like` has been used.

2) So how will the `Deparser` and/or `toString()` method rebuild the original SQL statement correctly?
```java
public String getStringExpression() {
        -- something needs to be done for RLIKE here
        return caseInsensitive ? "ILIKE" : "LIKE";
    }
```

3) Are you sure, your provided test case ran successfully? Have you tried `gradle build` or `mvn package` before submitting?

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt [hunk 1]
```diff
@@ -3300,7 +3300,7 @@ Expression LikeExpression(Expression leftExpression) #LikeExpression:
     Expression escape;
 }
 {
-    [<K_NOT> { result.setNot(true); } ] ( <K_LIKE> | <K_ILIKE> { result.setCaseInsensitive(true); } ) rightExpression=SimpleExpression()
+    [<K_NOT> { result.setNot(true); } ] ( <K_LIKE> | <K_ILIKE> | <K_RLIKE> { result.setCaseInsensitive(true); } ) rightExpression=SimpleExpression()
```
Comment (manticore-projects) [inline src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:3303]: Unfortunately, you break the Logic here:
1) For `<K_LIKE>` only, `caseInsensitive:=false` and
2) For `<K_ILIKE>` only,   `caseInsensitive:=false` 

You set `caseInsensitive` for `<K_ILKE>` instead.

**Comment by manticore-projects** [comment]
Incorporated via https://github.com/manticore-projects/JSqlParser/commit/f132547f56a1edd6e6a0e1454b53ac2e422a835b

**Pull request closed without merging**